apps/rewards/src/invite/rewardSender.service.ts

```diff
@@ -0,0 +1,156 @@
+import { KomenciLoggerService } from '@app/komenci-logger'
+import { ContractKit } from '@celo/contractkit'
+import { Inject, Injectable } from '@nestjs/common'
+import { Cron, CronExpression } from '@nestjs/schedule'
+import BigNumber from 'bignumber.js'
+import { EntityManager, In, Raw, Repository } from 'typeorm'
+import { appConfig, AppConfig } from '../config/app.config'
+import { RelayerProxyService } from '../relayer/relayer_proxy.service'
+import { InviteReward, RewardStatus } from './inviteReward.entity'
+import { InviteRewardRepository } from './inviteReward.repository'
+
+const WEI_PER_UNIT = 1000000000000000000
+
+interface WatchedInvite {
+  inviteId: string
+  txHash: string
+  sentAt: number
+  markAsDeadLetterIfFailed: boolean
+}
+
+@Injectable()
+export class RewardSenderService {
+  watchedInvites: Set<WatchedInvite>
+
+  constructor(
+    private readonly inviteRewardRepository: InviteRewardRepository,
+    private readonly relayerProxyService: RelayerProxyService,
+    @Inject(appConfig.KEY)
+    private readonly appCfg: AppConfig,
+    private readonly contractKit: ContractKit,
+    private readonly logger: KomenciLoggerService
+  ) {
+    this.watchedInvites = new Set()
+  }
+
+  async wasTxSentSuccesfully(txHash: string) {
+    const tx = await this.contractKit.web3.eth.getTransaction(txHash)
+    return Boolean(tx.blockHash)
+  }
+
+  @Cron(CronExpression.EVERY_SECOND)
+  async checkWatchedInvites() {
+    await Promise.all(
```

Comment:
We can use `Promise.allSettled` to allow all checks to continue in the case that one fails. For example, `wasTxSentSuccessfully` will throw an exception in the case that the tx not included in a block.

Reply:
Yes, great catch, I didn't remember `allSettled` existed but this is the perfect use case for it.

Reply:
An important note is that `Promise.allSettled` [requires node 12.9 or above](https://nodejs.org/en/blog/release/v12.9.0/) and we use node 10 in other repos, just to keep in mind.